kolibri/plugins/learn/assets/src/composables/useLearningActivities.js

```diff
@@ -8,18 +8,18 @@ import { get } from '@vueuse/core';
 import { LearningActivities } from 'kolibri.coreVue.vuex.constants';
 import { coreStrings } from '../../../../../core/assets/src/mixins/commonCoreStrings';
```

Comment:
Sorry, I missed this in the first pass - we shouldn't import core API objects in this way, as otherwise this will replicate this code into the learn plugin (using the import below, it accesses it at runtime via the default frontend object).
```suggestion
import coreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
```

Reply:
This ends up with the following Webpack error

```
[Vue warn]: Error in data(): "TypeError: kolibri_coreVue_mixins_commonCoreStrings__WEBPACK_IMPORTED_MODULE_3___default.a.$tr is not a function"
```

I think this may be happening because `commonCoreStrings` [exports Vue mixin by default](https://github.com/MisRob/kolibri/blob/f9fcd0a6f4e6315ffa6e0af64958e7d020cbf495/kolibri/core/assets/src/mixins/commonCoreStrings.js#L1089) that is then registered [in the core API](https://github.com/MisRob/kolibri/blob/f9fcd0a6f4e6315ffa6e0af64958e7d020cbf495/kolibri/core/assets/src/core-app/apiSpec.js#L48). I'd like to only use [this translator](https://github.com/MisRob/kolibri/blob/f9fcd0a6f4e6315ffa6e0af64958e7d020cbf495/kolibri/core/assets/src/mixins/commonCoreStrings.js#L8) though. Is that a problem too? I think I already saw translator functions being imported and used in javascript files.

Reply:
It's fine to import translators across files, it's just a special case for the core API. The only issue is that it will bundle the code for the core translator into the learn plugin, and then the string extraction machinery will duplicate the core strings into the learn plugin message bundles too, so it will cause a fair amount of bloat.

I think maybe better to add just the strings translator to the core API?

Can update this import https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/assets/src/core-app/apiSpec.js#L48 to import `commonCoreStrings, { coreStrings }` and then add `coreStrings` to the core API here under utils? https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/assets/src/core-app/apiSpec.js#L218

Reply:
I see now, thank you. Yes, I can add it to the core utils.

Reply:
Only thing to be wary about here is that in tests you may need to explicitly mock the `coreString` import in this case, as our Jest auto-import patching for coreAPI imports only works for default exports.

Reply:
`import commonCoreStrings, { coreStrings } from '../mixins/commonCoreStrings';` is breaking the build process:

```
Error: Cannot find module '/home/michaela/dev/kolibri/packages/kolibri-tools/dist/apiSpec.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/michaela/dev/kolibri/packages/kolibri-tools/lib/apiSpecExportTools.js:106:13)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/michaela/dev/kolibri/packages/kolibri-tools/lib/webpack.config.base.js:20:23)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
internal/modules/cjs/loader.js:638
    throw err;
    ^
```

and the linter as well:

```
kolibri/plugins/user_auth/assets/test/views/auth-select-page.spec.js
  1:1   error    Resolve error: Cannot find module '/home/michaela/dev/kolibri/packages/kolibri-tools/dist/apiSpec.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/home/michaela/dev/kolibri/packages/kolibri-tools/lib/apiSpecExportTools.js:106:13)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)  import/namespace
```

I wasn't able to find out what's the problem in a reasonable timeframe so I just left those two imports on different lines.

Reply:
Should be resolved and will follow-up on Slack with modules resolution errors

Reply:
Yeah that seems fine, probable that some AST parsing code is not setup to handle the same line import.